torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -759,6 +759,42 @@ def test_profiler_with_remote_builtin(self):
             use_record_function=True
         )
 
+    @dist_init
+    def test_profiler_with_script_async_rpc(self):
+        self._profiler_test_with_rpc(
+            RPCExecMode.ASYNC, my_script_func, args=(torch.tensor(1),)
+        )
+        self._profiler_test_with_rpc(
+            RPCExecMode.ASYNC,
+            my_script_func,
+            args=(torch.tensor(1),),
+            use_record_function=True,
+        )
```

Comment:
Shouldn't we add a unit test where rpc_async is called within a torchscript function and verify that is recorded appropriately as well?

Reply:
This is a case where profiling isn't supported, since this diff doesn't touch the code path where rpc_async is called from within a TS function.